.github/workflows/pull_request.yml

```diff
@@ -109,3 +109,36 @@ jobs:
       - uses: actions/checkout@v4
       - name: Check Spelling
         uses: crate-ci/typos@master
+
+  e2e:
+    name: e2e_kube
+    runs-on: ubuntu-22.04
+    strategy:
+      fail-fast: false
+      matrix:
+        k8s:
+          - v1.27.3
+          - v1.28.7
+          - v1.29.2
+    steps:
+      - uses: actions/checkout@v4
+      - name: Create kubernetes cluster with kind
+        env:
+          K8S_VERSION: ${{ matrix.k8s }}
+        run: |
+          ./ci/scripts/kind.sh
+
+      - name: Export logs
+        if: always()
+        run: |
+          mkdir -p _artifacts
+          docker ps
+          kind export logs  --loglevel=debug ./_artifacts/logs
+          ls _artifacts
+
+      - name: Upload logs
+        if: always()
```

Comment:
> https://github.com/xline-kv/Xline/pull/403#discussion_r1525719154
> I think uploading detailed logs when deployment failed is a better choice.

@Phoenix500526 

I think this would still be of value, at least proving that the log collection functionality works, but if you still prefer to have it only work on failure, I'd be happy to update it.

Reply:
I prefer to have it on failure. In general, we only check the logs if a task fails. So why upload logs when it succeeds? Additionally, the success of log uploads has no bearing on Xline itself and doesn't require repeated verification.

Reply:
Updated.